### PR TITLE
Refactor to use standard `Problem` for error response.

### PR DESCRIPTION
**Pull request by jogura**
* Refactor ErrorResponseBuilder to use new Problem
* Refactor Resources to use modified ErrorResponseBuilder
* Delete original ErrorResponse class
* Add HalProblem wrapper to produce the correct Swagger response in the UI

I just saw @drsnyder comment regarding using Jersey's built-in exception mappers to handle responses. I can refactor this code again if needed based on the results of that work.

RFCR @jimmordino @armonr

### CODE REVIEW

File: service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java [hunk 1]
```diff
@@ -79,6 +78,7 @@ public Response getForums(
 
     Optional<ForumRoot> forumRoot = forumService.getForums(siteId.get());
 
+    System.out.println("testing5");
```
Comment (armonr) [inline service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java]: Debug statement?

**Comment by armonr** [comment]
minor comment. LGTM

File: service/discussion/src/main/java/com/wikia/discussionservice/domain/hal/output/HalProblem.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  public String getDetail() {
+    return problem.getDetail();
+  }
+
+  public String getInstance() {
+    return problem.getInstance().toString();
+  }
+}
```
Comment (drsnyder) [inline service/discussion/src/main/java/com/wikia/discussionservice/domain/hal/output/HalProblem.java]: This is required for swagger?

Comment (jogura) [inline service/discussion/src/main/java/com/wikia/discussionservice/domain/hal/output/HalProblem.java]: Yes, in the class's javadoc I referred to the fact the [overriding models technique](`https://github.com/swagger-api/swagger-core/wiki/overriding-models`) doesn't seem to work with our version of `dropwizard-swagger` so in order to produce the correct documentation this was the work-around.

File: service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java [hunk 3]
```diff
@@ -177,9 +187,13 @@ public Response deleteForum(
...
 
-    return ErrorResponseBuilder.buildErrorResponse(10101, String.format(
-        "No forum found for site id: %s with forum id: %s", siteId.get(),
-        forumId.get()), null, Response.Status.NOT_FOUND);
+    return ErrorResponseBuilder.buildErrorResponse(
+        String.format("No forum found for site id: %s with forum id: %s", siteId.get(),
+                      forumId.get()),
+        null,
+        Response.Status.NOT_FOUND,
+        null,
```
Comment (cmiller-wikia) [inline service/discussion/src/main/java/com/wikia/discussionservice/resources/ForumResource.java]: Methods where half the arguments are null half the time make me nervous. I'd much rather have a version of the method that explicitly didn't accept those arguments.

**Comment by drsnyder** [comment]
:eyes: :+1: TM

**Pull request merged**